java/android/hello/src/main/java/com/zeroc/hello/HelloApp.java

```diff
@@ -104,18 +103,23 @@ else if(m.what == MSG_EXCEPTION || m.what == MSG_RESPONSE)
                 KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
                 keyStore.load(getResources().openRawResource(R.raw.client), "password".toCharArray());
 
+                // Create a key manager manager factory that uses the certificates from our KeyStore.
+                KeyManagerFactory keyManagerFactory =
+                    KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+                keyManagerFactory.init(keyStore, "password".toCharArray());
```

Comment:
Required for the client to provide a certificate, the hello server requires a client certificate by default.